modules/core/src/test/java/org/apache/ignite/internal/processors/diagnostic/DiagnosticProcessorTest.java

```diff
@@ -167,24 +192,237 @@ public void testWalDirs() throws Exception {
     }
 
     /**
-     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt"
-     * will be created and a warning will be in the log.
+     * Simple POJO to generate binary metadata in {@link #doTestOutputDiagnosticCorruptedPagesInfo(Consumer, BiConsumer)}
+     */
+    public static class TestValue {
+        public final String val;
+
+        public TestValue(String val) {
+            this.val = val;
+        }
+    }
+
+    /**
+     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt" will be created, data will
+     * be dumped and a warning will be in the log.
      *
      * @throws Exception If failed.
      */
     @Test
     public void testOutputDiagnosticCorruptedPagesInfo() throws Exception {
+        doTestOutputDiagnosticCorruptedPagesInfo(cfg -> {}, (n, anyPageId) -> {
+            validateDiagnosticPathDir(n);
+
+            File baseDumpDir = getBaseDumpDir(n);
+            assertEquals(7, baseDumpDir.list().length);
+
+            // 6 folders to validate.
+            validateCacheDir(anyPageId, baseDumpDir);
+            validateWalDir(baseDumpDir);
+            validateBinaryMetaDir(baseDumpDir);
+            validateCpDir(baseDumpDir);
+            validateMetaStorageDir(baseDumpDir);
+            validateUtilityCacheDir(baseDumpDir);
+            validateLogsCacheDir(baseDumpDir);
+        });
+    }
+
+    /**
+     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt" will be created, data will
+     * be dumped and a warning will be in the log.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testOutputDiagnosticCorruptedPagesInfoEncryptionEnabled() throws Exception {
+        Consumer<IgniteConfiguration> cfgClosure = cfg -> {
+            KeystoreEncryptionSpi encSpi = new KeystoreEncryptionSpi();
+
+            encSpi.setKeyStorePath(KEYSTORE_PATH);
+            encSpi.setKeyStorePassword(KEYSTORE_PASSWORD.toCharArray());
+
+            cfg.setEncryptionSpi(encSpi);
+        };
+
+        doTestOutputDiagnosticCorruptedPagesInfo(cfgClosure, (n, anyPageId) -> {
+            validateDiagnosticPathDir(n);
+
+            File baseDumpDir = getBaseDumpDir(n);
+            assertEquals(8, baseDumpDir.list().length);
+
+            // 7 folders to validate.
+            validateCacheDir(anyPageId, baseDumpDir);
+            validateWalDir(baseDumpDir);
+            validateBinaryMetaDir(baseDumpDir);
+            validateCpDir(baseDumpDir);
+            validateMetaStorageDir(baseDumpDir);
+            validateUtilityCacheDir(baseDumpDir);
+            validateLogsCacheDir(baseDumpDir);
```

Comment:
This is quite a duplication, not just 3 lines. I suggest extracting this to something like `validateCommonDirs()`

Reply:
Is it worth it? Right now it's explicit at least, and it's only duplicated 2 times

Reply:
Let's imagine we want to add another common directory. We add it in one place, but forget about the second one. And it will not be easy to spot that something is missing because 7 lines is quite a lot

Reply:
Ok